### PR TITLE
RSDK-7408 - Avoid module data dir conflicts

### PR DESCRIPTION
**Pull request by cheukt**
Based on logs and investigation, pretty sure what happened was that a separate test deleted the module data directory as we were making it, so I think the best move here is to make temp dirs for all module data dirs.

* Adds a local robot option to set viam home dir, which will change the module parent data directory - the only other place that uses `config.ViamDotDir` is the camera logger, but not easy to propagate changes to there. Happy to change var name to `modHomeDir` or something more specific if wanted.
* Also for the cases where we start viam-server as a managed process, added a util to automatically update the home dir

### CODE REVIEW

**Review by zaporter-work — APPROVED** [review]
LGTM! Nice catch

**Comment by zaporter-work** [comment]
(For context, I am not 100% certain that this is related to discovery_test and worry that the data dir logs might be a bit of a red herring, but this is a great cleanup either way)

File: examples/customresources/demos/multiplemodules/moduletest/module_test.go [hunk 1]
```diff
@@ -39,18 +38,7 @@ func TestMultipleModules(t *testing.T) {
...
-		serverPath := testutils.BuildTempModule(t, "web/cmd/server/")
-
-		// start the viam server with a temporary home directory so that it doesn't collide with
-		// the user's real viam home directory
-		testTempHome := t.TempDir()
-		server := pexec.NewManagedProcess(pexec.ProcessConfig{
-			Name:        serverPath,
-			Args:        []string{"-config", cfgFilename},
-			CWD:         utils.ResolveFile("./"),
-			Environment: map[string]string{"HOME": testTempHome},
```
Comment (dgottlieb) [inline examples/customresources/demos/multiplemodules/moduletest/module_test.go:51]: I assume these tests that had the temp home already were changed as part of a refactor? And the other tests that omitted the `Environment` configuration were the ones conflicting?

**Review by dgottlieb — APPROVED** [review]

File: examples/customresources/demos/multiplemodules/moduletest/module_test.go (hunk 1 above)
Comment (cheukt) [reply examples/customresources/demos/multiplemodules/moduletest/module_test.go:51]: yep! Specifically the test that failed was https://github.com/viamrobotics/rdk/blob/a3e8dd471b6a63afbb951a73664d7f68f9d78e0e/robot/impl/discovery_test.go#L117, that uses the `setupLocalRobot` helper

**Pull request merged**